src/transformStateWithClones.js

```diff
@@ -7,7 +7,53 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clones = [];
+  const stateClone = { ...state };
+  // i like my way
+  // for (let i = 0; i < actions.length; i++) {
+  //   if (actions[i].type === 'addProperties') {
+  //     Object.assign(stateClone, actions[i].extraData);
+  //     clones[i] = Object.assign({}, stateClone);
+  //   } else if (actions[i].type === 'removeProperties') {
+  //     for (let j = 0; j < actions[i].keysToRemove.length; j++) {
+  //       const keyToRemove = actions[i].keysToRemove[j];
+
+  //       delete stateClone[keyToRemove];
+  //     }
+  //     clones[i] = Object.assign({}, stateClone);
+  //   } else if (actions[i].type === 'clear') {
+  //     for (const k in stateClone) {
+  //       delete stateClone[k];
+  //     }
+  //     clones[i] = Object.assign({}, stateClone);
+  //   };
+  // }
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        clones.push(Object.assign({}, stateClone));
```

Comment:
let's move `push` out of the switch, and just push it once at the end of the loop